refactor: refactor flat search to lance-index (#1419)
lancedb/lance: rust/lance-index/README.md

# lance-index

> `lance-index` is a internal sub-crate, containing various vector index implementations
> used the [Lance](https://github.com/lancedb/lance).

Reviewer: ```suggestion
`lance-index` is an internal sub-crate, containing various vector index implementations
used in [Lance](https://github.com/lancedb/lance).
```
